packages/story-editor/src/components/canvas/mediaRecordingLayer.js

```diff
@@ -0,0 +1,261 @@
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import styled from 'styled-components';
+import { __ } from '@googleforcreators/i18n';
+import {
+  useCallback,
+  useEffect,
+  usePrevious,
+  useRef,
+  useDebouncedCallback,
+} from '@googleforcreators/react';
+import {
+  blobToFile,
+  createBlob,
+  getImageFromVideo,
+} from '@googleforcreators/media';
+import { format } from '@googleforcreators/date';
+
+/**
+ * Internal dependencies
+ */
+import {
+  useMediaRecording,
+  PermissionsDialog,
+  ErrorDialog,
+  Countdown,
+  VideoMode,
+  ProgressBar,
+  DurationIndicator,
+  Footer,
+  SettingsModal,
+  LayerWithGrayout,
+  DisplayPageArea,
+  Wrapper,
+} from '../mediaRecording';
+import { PageTitleArea } from './layout';
+
+const VideoWrapper = styled.div`
+  width: 100%;
+  height: 100%;
+  background: #000;
+  margin: 0 auto;
+  display: flex;
+  align-items: center;
+`;
+
+const Video = styled.video`
+  display: block;
+  width: 100%;
+  height: 100%;
+  object-fit: cover;
+  border-radius: 5px;
+`;
+
+const Photo = styled.img`
+  display: block;
+  width: 100%;
+  height: 100%;
+  object-fit: cover;
+  border-radius: 5px;
+`;
+
+function MediaRecordingLayer() {
+  const {
+    updateMediaDevices,
+    status,
+    mediaBlob,
+    mediaBlobUrl,
+    setMediaBlobUrl,
+    liveStream,
+    getMediaStream,
+    setFile,
+    needsPermissions,
+    hasAudio,
+    isGif,
+    toggleIsGif,
+    isImageCapture,
+    audioInput,
+    videoInput,
+  } = useMediaRecording(({ state, actions }) => ({
+    status: state.status,
+    liveStream: state.liveStream,
+    mediaBlob: state.mediaBlob,
+    mediaBlobUrl: state.mediaBlobUrl,
+    hasAudio: state.hasAudio,
+    isGif: state.isGif,
+    needsPermissions:
+      ('idle' === state.status || 'acquiring_media' === state.status) &&
+      !state.videoInput,
+    audioInput: state.audioInput,
+    videoInput: state.videoInput,
+    isImageCapture: Boolean(state.file?.type?.startsWith('image')),
+    toggleIsGif: actions.toggleIsGif,
+    updateMediaDevices: actions.updateMediaDevices,
+    getMediaStream: actions.getMediaStream,
+    setFile: actions.setFile,
+    setMediaBlobUrl: actions.setMediaBlobUrl,
+  }));
+
+  const streamRef = useRef();
+  const videoRef = useRef();
+
+  const isReady = 'ready' === status;
+  const isFailed = 'failed' === status;
+
+  const previousAudioInput = usePrevious(audioInput);
+  const previousVideoInput = usePrevious(videoInput);
+
+  const isMuted = !hasAudio || isGif;
+
+  const captureImage = useCallback(async () => {
+    if (!streamRef.current) {
+      return;
+    }
+
+    const blob = await getImageFromVideo(streamRef.current);
+    try {
+      setMediaBlobUrl(createBlob(blob));
+    } catch (e) {
+      // Do nothing.
+    }
+
+    const imageFile = blobToFile(
+      blob,
+      `image-capture-${format(new Date(), 'Y-m-d-H-i')}.jpeg`,
+      'image/jpeg'
```

Comment:
We could use `MEDIA_POSTER_IMAGE_MIME_TYPE` here.

Reply:
It‘s not a poster image though, it‘s a fdifferent kind of image file that just happens to be a JPEG as well. Using this const would be misleading, which is why I didn‘t use it.